iceoryx_posh/include/iceoryx_posh/internal/popo/base_server.hpp

```diff
@@ -0,0 +1,156 @@
+    ///
+    bool hasRequests() const noexcept;
+
+    ///
+    /// @brief Check if requests has been missed since the last call of this method.
+    /// @return True if requests has been missed.
+    /// @details Requests may be missed due to overflowing receive queue.
+    ///
+    bool hasMissedRequests() noexcept;
+
+    /// @brief Releases any unread queued requests.
+    void releaseQueuedRequests() noexcept;
+
+    friend class NotificationAttorney;
+
+  protected:
+    using SelfType = BaseServer<Port>;
+    using PortType = Port;
+
+    BaseServer(const capro::ServiceDescription& service, const ServerOptions& serverOptions) noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Invalidates the internal triggerHandle.
+    /// @param[in] uniqueTriggerId the id of the corresponding trigger
+    void invalidateTrigger(const uint64_t uniqueTriggerId) noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Attaches the triggerHandle to the internal
+    /// trigger.
+    /// @param[in] triggerHandle rvalue reference to the triggerHandle. This class takes the ownership of that handle.
+    /// @param[in] serverState the state which should be attached
+    void enableState(iox::popo::TriggerHandle&& triggerHandle, const ServerState serverState) noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Returns method pointer to the event
+    /// corresponding hasTriggered method callback
+    /// @param[in] serverState the state to which the hasTriggeredCallback is required
+    WaitSetIsConditionSatisfiedCallback
+    getCallbackForIsStateConditionSatisfied(const ServerState serverState) const noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Resets the internal triggerHandle
+    /// @param[in] serverState the state which should be detached
+    void disableState(const ServerState serverState) noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Attaches the triggerHandle to the internal
+    /// trigger.
+    /// @param[in] triggerHandle rvalue reference to the triggerHandle. This class takes the ownership of that handle.
+    /// @param[in] serverEvent the event which should be attached
+    void enableEvent(iox::popo::TriggerHandle&& triggerHandle, const ServerEvent serverEvent) noexcept;
+
+    /// @brief Only usable by the WaitSet/Listener, not for public use. Resets the internal triggerHandle
+    /// @param[in] serverEvent the event which should be detached
+    void disableEvent(const ServerEvent serverEvent) noexcept;
+
+    ///
+    /// @brief port
+    /// @return const accessor of the underlying port
+    ///
+    const Port& port() const noexcept;
+
+    ///
+    /// @brief port
+    /// @return accessor of the underlying port
+    ///
+    Port& port() noexcept;
+
+    Port m_port{nullptr};    // NOLINT(cppcoreguidelines-non-private-member-variables-in-classes)
```

Comment:
Same thing like in the `base_client`. How can you use a `nullptr` here when the constructor requires a mutable reference?

Reply:
@elfenpiff because I'm a chicken and want to save my future self which might add another cTor and forget to set `m_port`. In this case it is at least initialized with a `nullptr`. It is a habit from me to always initialize with `nullptr` even if the cTor will immediately override it. But in this case I could also just use a reference

Reply:
@elfenpiff forget what I wrote, I misunderstood you